crypto.go

```diff
@@ -53,3 +80,121 @@ func GenerateSigningKeys() (SigningKeys, error) {
 	}
 	return signingKeys, err
 }
+
+// EncryptData accepts a record object of plain meta and plain data, returning
+// a record with data keys encrypted with dk and ak according to Tozstore spec.
+func EncryptData(Data map[string]string, ak SymmetricKey) *map[string]string {
+	encryptedData := make(map[string]string)
+	for k, v := range Data {
+		dk := RandomSymmetricKey()
+		ef, efN := SecretBoxEncryptToBase64([]byte(v), dk)
+		edk, edkN := SecretBoxEncryptToBase64(dk[:], ak)
+		encryptedData[k] = fmt.Sprintf("%s.%s.%s.%s", edk, edkN, ef, efN)
+	}
+	return &encryptedData
+}
+
+// EncryptAccessKey returns encrypted access key with nonce attached.
+func EncryptAccessKey(rawAK SymmetricKey, encryptionKeys EncryptionKeys) (string, error) {
+	eak, eakN, err := BoxEncryptToBase64(rawAK[:], encryptionKeys)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s.%s", eak, eakN), err
+}
+
+// RandomSymmetricKey generates a random symmetric key (secret).
+func RandomSymmetricKey() SymmetricKey {
+	key := &[SymmetricKeySize]byte{}
+	_, err := rand.Read(key[:])
+	if err != nil {
+		// we don't expect this to fail
+		panic("random number generation failed")
+	}
+	return key
+}
+
+// RandomNonce generates a random nonce of size NoneSize
+func RandomNonce() Nonce {
+	b := [NonceSize]byte{}
+	_, err := rand.Read(b[:])
+	if err != nil {
+		// we don't expect this to fail
+		panic("random number generation failed")
+	}
+	return &b
```

Comment:
Addressof calls are at different places in RandomNonce() and RandomSymmetricKey() -- is this intentional?